include/net/download_client.h

```diff
@@ -97,6 +97,10 @@ struct download_client_cfg {
 	 *  or NULL to use the default APN.
 	 */
 	const char *apn;
+	/** Maximum fragment size to download. 0 indicates that Kconfigured
+	 *  values shall be used.
+	 */
+	size_t max_fragment_size;
```

Comment:
The idea is that `struct download_client_cfg` contains download-specific configs that the user might change between downloads, while `struct download_client` contains the lib's "private" state. In this case, it is appropriate to have the custom fragment size in `struct download_client_cfg` since the user might change that between downloads. If possible please rename to `frag_size_override`.